airbyte-ci/connectors/base_images/base_images/python/v1.py

```diff
@@ -0,0 +1,79 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+"""This module declares all the airbyte python connector base image for version 1.
+Please create a v2.py module if you want to declare a new major version.
+"""
+from __future__ import annotations
+
+from typing import Final, final
+
+import dagger
+from base_images import common, sanity_checks
+from base_images.python.common import AirbytePythonConnectorBaseImage, PythonBase
+
+
+class _1_0_0(AirbytePythonConnectorBaseImage):
+    base_base_image: Final[PythonBase] = PythonBase.PYTHON_3_9_18
+
+    changelog_entry: Final[
+        str
+    ] = "Declare our first base image version. It uses Python 3.9.18 on a Debian 11 (Bookworm) system with Pip 23.2.1 and UTC timezone."
+
+    run_previous_version_sanity_checks = False
+
+    @property
+    def container(self) -> dagger.Container:
+        pip_cache_volume: dagger.CacheVolume = self.dagger_client.cache_volume(AirbytePythonConnectorBaseImage.pip_cache_name)
+
+        return (
+            self.base_container.with_mounted_cache("/root/.cache/pip", pip_cache_volume)
+            # Set the timezone to UTC
+            .with_exec(["ln", "-snf", "/usr/share/zoneinfo/Etc/UTC", "/etc/localtime"])
+            # Upgrade pip to the expected version
+            .with_exec(["pip", "install", "--upgrade", "pip==23.2.1"])
+        )
+
+    @final
+    @staticmethod
+    async def run_sanity_checks(base_image_version: common.AirbyteConnectorBaseImage):
+        await sanity_checks.check_timezone_is_utc(base_image_version.container)
+        await sanity_checks.check_a_command_is_available_using_version_option(base_image_version.container, "bash")
+        await sanity_checks.check_python_version(base_image_version.container, "3.9.18")
+        await sanity_checks.check_pip_version(base_image_version.container, "23.2.1")
+
+
+class _1_1_0(AirbytePythonConnectorBaseImage):
```

Comment:
@erohmensing I think I can explain why I naturally came up wit this pattern of hard coding versions. It reminds me my terraform background: I find it beautiful to declare resources as code.  I believe base versions are resources because they are consumed, through the GLOBAL_REGISTRY instance, by our connector build pipeline.
We have to maintain a lively list of our released image in these module as multiple version 
can coexist because different connector can use different base versions 
If we follow the terraform metaphore an `apply` command would publish these image to dockerhub.
The main difference here is that:
- terraform has a notion of `state` that we don't have here. I believe its something I should add. Basically a file listing all versions that is generated and that we (or a bot) can compare on module change to make sure the changes are correct and that previous versions are not mutated.

I'd understand if you consider all this a bit far fetched. We can without much effort just make the AirbytePythonConnectorBaseImage not abstract and set a version attribute on it.